src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -592,7 +594,13 @@ public void startSearchMatchmaker() {
         .toFuture()
         .thenAccept(featuredModBean -> updateGameIfNecessary(featuredModBean, Set.of()))
         .thenCompose(aVoid -> fafServerAccessor.startSearchMatchmaker())
-        .thenCompose(gameLaunchResponse -> downloadMapIfNecessary(gameLaunchResponse.getMapName()).thenCompose(aVoid -> leaderboardService.getActiveLeagueEntryForPlayer(playerService.getCurrentPlayer(), gameLaunchResponse.getLeaderboard()))
+        .thenCompose(gameLaunchResponse -> downloadMapIfNecessary(gameLaunchResponse.getMapName()).thenCompose(aVoid -> {
+              if (isRunning()) {
+                gameKilled = true;
+                process.destroy();
+              }
```

Comment:
I have no idea how I would test this in a unit test

Reply:
I guess we would then have to differentiate running a replay and running a game. We still want to prevent joining the queue when you are in a custom game lobby. We could also restrict to only one replay running this way.
Should be easy to implement now that I think about it